sonarqube/resource_sonarqube_setting.go

```diff
@@ -270,26 +271,35 @@ func synchronizeSettings(d *schema.ResourceData, m interface{}) (bool, error) {
 	apiComponentSettings, _ := getComponentSettings(componentId, m)
 
 	// Make sure the order is always the same for when we are comparing lists of conditions
-	sort.Slice(componentSettings, func(i, j int) bool {
-		return componentSettings[i].(map[string]interface{})["key"].(string) < componentSettings[j].(map[string]interface{})["key"].(string)
-	})
+	// sort.Slice(componentSettings, func(i, j int) bool {
+	// 	return componentSettings[i].(map[string]interface{})["key"].(string) < componentSettings[j].(map[string]interface{})["key"].(string)
+	// })
```

Comment:
I'm not sure this is a good idea - a) there is no place for commented out code IMO - if you want to remove it then remove it!
b) If the settings come back in a different order then you will end up replacing them unnecessarily - ISTR on one of my PRs (it may even have been one introducing this code) that I had to ensure order for precisely this reason - that Sonar did not return them in the order I added them.  We certainly can't assume people will add settings in the "right" order that a GET will return them in the same order too.